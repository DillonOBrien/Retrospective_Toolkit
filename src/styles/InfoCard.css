.infoCard--root {
  width: 180%;
  display: flex;
  flex-direction: column;
  justify-content: left;
  margin: auto;
  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.2);
  transition: all 0.3s cubic-bezier(0.25, 0.8, 0.25, 1);
  overflow: hidden;
}

.title-button {
  padding-top: 13px;
  text-align: center;
  width: 180.5%;
  margin: auto;
  border: none;
  background-color: white;
  border: 1px solid rgba(0, 0, 0, 0.12);
  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.2);
  transition: all 0.3s cubic-bezier(0.25, 0.8, 0.25, 1);
}

.title-button:hover {
  box-shadow: 5 15px 20px rgba(0, 0, 0, 0.25);
  background-color: rgba(246, 248, 250, 0.356);
}

.title-button:focus {
  border: 4px dotted red;
  text-decoration: none;
  box-shadow: none;
}

.info-open {
  max-height: 400px;
  transition: max-height 0.4s ease;
  height: 88%;
}

.info-close {
  max-height: 0px;
  transition: max-height 0.4s ease;
}
