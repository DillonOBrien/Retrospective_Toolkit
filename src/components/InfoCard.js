/**
 * Base class of a text-based info card. It is expected to use a few paragraphs of text that will be
 * separated by line breaks.
 */
import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardActions from "@material-ui/core/CardActions";
import CardContent from "@material-ui/core/CardContent";
// import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";
import PropTypes from "prop-types";

// Custom CSS elements for an info card.
const useStyles = makeStyles({
  root: {
    width: "200%",
    height: "88%",
    maxHeight: "400px",
    display: "flex",
    flexDirection: "column",
    justifyContent: "left",
    margin: "10px 0px 0px 10px", // top right bottom left
    boxShadow: "0 1px 3px rgba(0,0,0,0.12)",
    transition: "all 0.3s cubic-bezier(.25,.8,.25,1)",
    "&:hover": {
      boxShadow: "0 10px 13px rgba(0,0,0,0.25)",
    },
  },
});

const InfoCard = (props) => {
  const classes = useStyles();
  const { title, body, body2, body3 } = props;

  return (
    <Card className={classes.root} variant="outlined">
      <CardContent>
        <Typography
          variant="h4"
          className={classes.title}
          color="textPrimary"
          gutterBottom
          align="center"
        >
          {title}
        </Typography>
        <Typography variant="body1" component="p">
          {body}
        </Typography>
        <Typography className={classes.pos} component="p">
          <br />
          {body2}
        </Typography>
        <Typography variant="body2" component="p" color="textSecondary">
          {body3}
        </Typography>
      </CardContent>
      <CardActions />
    </Card>
  );
};
InfoCard.propTypes = {
  title: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  body2: PropTypes.string.isRequired,
  body3: PropTypes.string.isRequired,
};

export default InfoCard;
