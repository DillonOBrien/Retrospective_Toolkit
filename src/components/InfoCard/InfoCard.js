/**
 * Base class of a text-based info card. It is expected to use a few paragraphs of text that will be 
 * separated by line breaks.
 */
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';

//Custom CSS elements for an info card.
const useStyles = makeStyles({
  root: {
    maxWidth: '500px',
    maxHeight: '600px',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'left',
    margin: 10,
  },
});

const InfoCard = (props) => {
  const classes = useStyles();

  return (
    <Card className={classes.root} variant="outlined">
      <CardContent>
        <Typography variant="h4" className={classes.title} color="textPrimary" gutterBottom align="center">
          {props.title}
        </Typography>
        <Typography variant="body" component="p">
          {props.body}
        </Typography>
        <Typography className={classes.pos} component="p">
          <br></br>
          {props.body2}
        </Typography>
        <Typography variant="body2" component="p" color="textSecondary">
          {props.body3}
        </Typography>
      </CardContent>
      <CardActions>
        <Button size="small" onClick={props.clicked}>Learn More</Button>
      </CardActions>
    </Card>
  )
}

export default InfoCard;