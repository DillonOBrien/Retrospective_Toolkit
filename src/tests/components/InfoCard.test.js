/* eslint-disable no-console */
import React from "react";
import { shallow } from "enzyme";
import InfoCard from "../../components/InfoCard";

let wrapper;

beforeEach(() => {
  jest.spyOn(console, "warn").mockImplementation(() => {});
  jest.spyOn(console, "error").mockImplementation(() => {});
  jest.spyOn(console, "log").mockImplementation(() => {});
  wrapper = shallow(<InfoCard />);
});

describe("Components/InfoCard Testing", () => {
  // Snapshot for InfoCard
  test("SNAPSHOT: InfoCard", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
