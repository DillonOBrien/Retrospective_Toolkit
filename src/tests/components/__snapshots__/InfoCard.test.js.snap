// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/InfoCard Testing SNAPSHOT: InfoCard 1`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-root-1"
  variant="outlined"
>
  <WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(Typography))
      align="center"
      color="textPrimary"
      gutterBottom={true}
      variant="h4"
    />
    <WithStyles(ForwardRef(Typography))
      component="p"
      variant="body1"
    />
    <WithStyles(ForwardRef(Typography))
      component="p"
    >
      <br />
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      color="textSecondary"
      component="p"
      variant="body2"
    />
  </WithStyles(ForwardRef(CardContent))>
  <WithStyles(ForwardRef(CardActions)) />
</WithStyles(ForwardRef(Card))>
`;
